backend/src/main/java/com/bakdata/conquery/models/events/CBlock.java

```diff
@@ -359,18 +365,27 @@ private static CDateRange[] calculateEntityDateIndices(Bucket bucket, int bucket
 	 * Helper method for calculateEntityDateIndices, swapping {@link Integer#MIN_VALUE}/{@link Integer#MAX_VALUE} for performance.
 	 */
 	private static CDateRange calculateSpan(int max, int min, CDateRange in) {
+
 		if (max == Integer.MIN_VALUE && min == Integer.MAX_VALUE) {
 			return in;
 		}
 
+		final CDateRange span;
+
 		if (max == Integer.MIN_VALUE) {
-			return in.span(CDateRange.atLeast(min));
+			span =  CDateRange.atLeast(min);
+		}
+		else if (min == Integer.MAX_VALUE) {
+			span = CDateRange.atMost(max);
+		}
+		else {
+			span = CDateRange.of(min, max);
 		}
```

Comment:
kann man anstatt dessen nicht ein `CDateRange.of(min, max)` machen?  das hat doch "fast" die selbe Logik.

Ich weiß aber auch nicht warum hier `max == Integer.MIN_VALUE` getestet werden sollte

Reply:
min ist initialisiert mit Integer.MAX und umgekehrt, dass Math.min/Math.max einfacher im Code zu handaben ist

Reply:
Ich hab das ganze nochmal aufgeräumt, das geflippte ist nur noch möglich, wenn man gar nichts hat. war dadurch einfacher